File: gcc/others/lab/learned/trial/bubble_sort.c
#include<stdio.h>

int main()
{
    int i, j, n, temp, a[30];
    printf("\n\n\tBUBBLE SORT\n");
    printf("Enter the no. of elements: ");
    scanf("%d", &n);

    printf("\nEnter the elements\n");
    for(i=0;i<n;i++)
        scanf("%d", &a[i]);

    for(i=0;i<n;i++)
    {
        for(j=0;j<n-1;j++)
        {
            if(a[j]>a[j+1])
            {
                temp=a[j];
                a[j]=a[j+1];
                a[j+1]=temp;
            }
        }
    }

    printf("\n---> OUTPUT\n");
    for(i=0;i<n;i++)
        printf("\t%d", a[i]);
    
    printf("\n");

}

File: S3/OOP_Java Lab/others/lab/Learned1/bubble_sort.java
import java.util.*;
import java.io.*;

public class bubble_sort
{
    public static void main(String []args)
    {
        int i, j, temp, n;

        System.out.println("\n\n\tBUBBLE SORT\n");
        System.out.print("Enter the number of elements: ");
        Scanner sc=new Scanner(System.in);
        n=sc.nextInt();

        System.out.println("");
        System.out.println("Enter the elements: ");

        int A[] = new int[n];
        for(i=0;i<n;i++)
        {
            A[i]=sc.nextInt();
        }

        for(i=0;i<n;i++)
        {
            for(j=0;j<n-1;j++)
            {
                if(A[j]>A[j+1])
                {
                    temp=A[j];
                    A[j]=A[j+1];
                    A[j+1]=temp;
                }
            }
        }

        System.out.println("\n --- SORTED ARRAY ---");
        for(i=0;i<n;i++)
        {
            System.out.print("\t"+A[i]);
        }
        System.out.println("");

    }
}